### PR TITLE
add setTypoTolerance to SearchParameters

### PR DESCRIPTION
**Pull request by maxiloc**

### CODE REVIEW

File: src/SearchParameters/index.js [hunk 1]
```diff
@@ -182,6 +182,20 @@ SearchParameters.prototype = {
...
+   * typoTolerance setter
+   * Set the value of typoTolerance
+   * @method
+   * @param {boolean} b boolean new value of typoTolerance
+   * @return {SearchParameters}
+   */
+  setTypoTolerance : function setTypoTolerance( b ) {
+    return this.mutateMe( function( m ) {
+      m.typoTolerance = b;
+      m.page = 0;
```
Comment (bobylito) [inline src/SearchParameters/index.js]: Good catch to reset the current page!

File: src/SearchParameters/index.js [hunk 2]
```diff
@@ -182,6 +182,20 @@ SearchParameters.prototype = {
...
+
+  /**
+   * typoTolerance setter
+   * Set the value of typoTolerance
+   * @method
+   * @param {boolean} b boolean new value of typoTolerance
+   * @return {SearchParameters}
+   */
+  setTypoTolerance : function setTypoTolerance( b ) {
+    return this.mutateMe( function( m ) {
```
Comment (bobylito) [inline src/SearchParameters/index.js]: It's not a boolean. it's a string with the values "true", "false" , "min" or "strict" http://algolia.github.io/algoliasearch-helper-js/docs/SearchParameters.html#typoTolerance https://www.algolia.com/doc/rest_api#typoTolerance

Comment (redox) [inline src/SearchParameters/index.js]: But from a JS POV it can be a boolean OR a string: ```true || false || "min" || "strict"```

**Comment by bobylito** [comment]
:+1: Awesome! Thanks for your first contribution!

**Pull request merged**